### PR TITLE
Add missing Crime/Billing systems + relationships

### LINKED ISSUE #35: Add Fee Calculator system

### LINKED ISSUE #37: Add HMCTS legacy applications
We currently have the Common Platform system documented at its interaction with the Court Data Adapter.

We should add:

- LIBRA
- INFOX - for MAAT/MLRA interaction with LIBRA
- XHBIT Live
- XHBIT Portal

### PR DESCRIPTION
**Pull request by richrace**
## What does this pull request do?

Adds in missing Systems, relationships and user journeys

Closes #35 
Closes #37 

## What is the intent behind these changes?

Take make sure the whole process is documented 

See the older diagrams for some context:
* https://media.githubusercontent.com/media/ministryofjustice/laa-architectural-diagrams/master/diagrams/get-paid/get-paid-for-crime-legal-aid-system-landscape.png
* https://media.githubusercontent.com/media/ministryofjustice/laa-architectural-diagrams/master/diagrams/get-legal-aid/criminal-legal-aid-application-system-landscape.png

## Preview

Using the Crime filter view: 

structurizr-61040-CRIME (3)

### CODE REVIEW

File: src/main/kotlin/model/CCCD.kt [hunk 1]
```diff
@@ -83,6 +83,13 @@ class CCCD private constructor() {
     }
 
     override fun defineRelationships() {
+      web.uses(
+        FeeCalculator.api,
+        "Uses API to calculator fees",
```
Comment (oliverweighell) [inline src/main/kotlin/model/CCCD.kt:88]: calculate

File: src/main/kotlin/model/FeeCalculator.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+class FeeCalculator private constructor() {
+  companion object : LAASoftwareSystem {
+    lateinit var system: SoftwareSystem
+    lateinit var api: Container
+
+    override fun defineModelEntities(model: Model) {
+      system = model.addSoftwareSystem(
+        "Fee Calculator",
+        "A system that keeps track of Fee Schemes and provides an API to query specific this large dataset"
```
Comment (oliverweighell) [inline src/main/kotlin/model/FeeCalculator.kt:17]: Not sure if 'specific' is needed?

File: src/main/kotlin/model/Allpay.kt [hunk 3]
```diff
@@ -27,6 +28,7 @@ class Allpay private constructor() {
     }
 
     override fun defineUserRelationships() {
+      system.uses(LegalAidAgencyUsers.provider, "Pays", null, null, tagsToArgument(Tags.CRIME))
```
Comment (adrianclay) [inline src/main/kotlin/model/Allpay.kt:31]: Interesting to know!

Comment (adrianclay) [reply src/main/kotlin/model/Allpay.kt:31]: In fact, this seems totally at odds with what I understand, perhaps I'm wrong.  Where did you get this info from?

Comment (richrace) [reply src/main/kotlin/model/Allpay.kt:31]: I was inferring this from https://media.githubusercontent.com/media/ministryofjustice/laa-architectural-diagrams/master/diagrams/get-paid/get-paid-for-crime-legal-aid-system-landscape.png

As the Allpay system states "Direct Debit payment processor"

Comment (richrace) [reply src/main/kotlin/model/Allpay.kt:31]: Might be something called "Bacway" instead.

File: src/main/kotlin/model/MLRA.kt [hunk 4]
```diff
@@ -55,6 +55,13 @@ class MLRA private constructor() {
     }
 
     override fun defineUserRelationships() {
+      LegalAidAgencyUsers.crimeApplicationCaseWorker.uses(
+        web,
+        "To record LAA decisions and send them to Magistrates Courts and to review Magistrates/Crown Court decisions",
```
Comment (adrianclay) [inline src/main/kotlin/model/MLRA.kt:60]: Could this "Send them to Magistrates Courts" be modelled as a relationship instead?

File: src/main/kotlin/model/MAAT.kt [hunk 5]
```diff
@@ -66,12 +77,27 @@ class MAAT private constructor() {
         null,
         tagsToArgument(Tags.CRIME)
       )
+
+      db.uses(
+        eForms.system,
```
Comment (adrianclay) [inline src/main/kotlin/model/MAAT.kt:82]: We've been modelling the HUB jobs as DB to DB communication. Is there a DB for `eForms`?

File: src/main/kotlin/model/Allpay.kt (hunk 3 above)
Comment (richrace) [reply src/main/kotlin/model/Allpay.kt]: Removed this assumption.

File: src/main/kotlin/model/MLRA.kt (hunk 4 above)
Comment (richrace) [reply src/main/kotlin/model/MLRA.kt]: Updated the wording on this.

**Review by adrianclay — APPROVED** [review]
It's a big PR, with lots of changes but all the issues I see are good now!

**Pull request merged**